lib/opentelemetry_absinthe.ex

```diff
@@ -1,7 +1,65 @@
 defmodule OpentelemetryAbsinthe do
+  alias OpentelemetryAbsinthe.Instrumentation
+
+  # Allow alias before moduledoc
+  # credo:disable-for-next-line
   @moduledoc """
-  OpentelemetryAbsinthe is an opentelemetry instrumentation library for Absinthe
+  # OpentelemetryAbsinthe
+  An opentelemetry instrumentation library for Absinthe
+
+  ## Usage
+
+  To start collecting traces just put `OpentelemetryAbsinthe.setup()` in your application start function.
+
+  ## Configuration
+
+  OpentelemetryAbsinthe can be configured with the application environment
+  ```
+  config :opentelemetry_absinthe,
+    trace_request_query: false,
+    trace_response_error: true
+  ```
+  configuration can also be passed directly to the setup function
+  ```
+  OpentelemetryAbsinthe.setup(
+    trace_request_query: false,
+    trace_response_error: true
+  )
+  ```
+
+  ## Configuration options
+    
+    * `span_name`(default: #{Instrumentation.default_config()[:span_name]}): the name of the span attributes will be attached to
+    * `trace_request_query`(default: #{Instrumentation.default_config()[:trace_request_query]}): attaches the graphql query as an attribute
```

Comment:
I didn't know this is possible and if it really works it's really cool :star_struck:

Reply:
Yeah it's really cool, nice way to ensure the docs stay up to date. 

semi-related but I first saw this pattern 2 days ago on another repo, but I couldn't remember what repo. I wanted to link it here for reference and because having everything documented makes my brain happy, so I decided to spend the 5 minutes trying to find it.

I wasted over an hour

it was redact_ex by the way
https://hexdocs.pm/redact_ex/RedactEx.Redactor.html#module-configuration

Reply:
Thanks for the reference :pray: